AWSAppSyncClient/AWSAppSyncClient.swift

```
@@ -184,9 +183,12 @@ public class AWSAppSyncClient {
         } catch {
```

Need to validate if Promise resolved to failure

---

await should throw the error if failure